sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/CommonFileDataSourceSuite.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources
+
+import org.scalatest.funsuite.AnyFunSuite
+
+import org.apache.spark.sql.{Encoders, FakeFileSystemRequiringDSOption, QueryTest, Row}
+import org.apache.spark.sql.catalyst.plans.SQLHelper
+import org.apache.spark.sql.test.SQLTestData
+
+// The trait contains tests for all file-based data sources.
+trait CommonFileDataSourceSuite {
+  self: QueryTest with AnyFunSuite with SQLTestData with SQLHelper =>
+
+  protected def dataSourceFormat: String
+
+  test(s"Propagate Hadoop configs from $dataSourceFormat options to underlying file system") {
```

Comment:
Shall we add this test into `FileBasedDataSourceSuite`?

Reply:
Do you mean add it to `FileBasedDataSourceSuite` and remove it from `JsonSuite`, `CSVSuite` and so on? In that case, we will not test both DS v1 and v2.

Reply:
If I place the test to `FileBasedDataSourceSuite`, will I be able to test Avro? It will look like slightly strange that we test `external/avro` in `sql/core`, won't it?

Reply:
Hmmm.. okay